### ISSUE TITLE
When starting an episode of series there are sound problems with the Disney+ addon

### ISSUE THREAD

**Issue by Ronny-nerd** [peak3d/inputstream.adaptive#1335]
### Describe the bug
Here is a clear and concise description of what the problem is:

Hardware: Nvidia Shield TV 2015 (Android TV11)
No AVR and no soundbar
InputStream Adative version 21.1.3

Kodi 21 Omega Alpha 2
With InputStream Adaptive version 21.1.0 and Kodi 20.2 Nexus with InputStream Adaptive version 20.3.11 there is not this problem.

When starting an episode of series there are sound problems with the Disney+ addon and for example with these series.

Secret Invasion
The Mandalorian
Obi-Wan Kenobi
The Book of Boba Fett
Star Wars: Andor
Star Wars: Tales of the Jedi

For me, there is no sound at the start of an episode in said series, only when you manually switch the sound.

Translated with www.DeepL.com/Translator (free version)

[Manifest-Problem 3 InputStream Adaptive.zip](https://github.com/xbmc/inputstream.adaptive/files/12122935/Manifest-Problem.3.InputStream.Adaptive.zip)

![072123115009](https://github.com/xbmc/inputstream.adaptive/assets/65504837/dcc93413-10dd-4b8d-a80f-57002b434843)

![072123115308](https://github.com/xbmc/inputstream.adaptive/assets/65504837/eef5eb52-bc4c-44c2-9bc1-5bc665c53f9e)

Sorry my english is no good.

**Comment by CastagnaIT** [comment]
you need to disable the dis€y+ manifest alteration proxy
and then provide new logs and manifests

**Comment by Ronny-nerd** [comment]
> Sie müssen den Manifeständerungs-Proxy von dis€y+ deaktivieren

And where can you set the Disney+ Manifest Proxy to disable in the Disney+ Addon?

**Comment by Ronny-nerd** [comment]
![072123164428](https://github.com/xbmc/inputstream.adaptive/assets/65504837/39499694-bbbe-4b24-8eee-4a435dfc709d)
![072123164535](https://github.com/xbmc/inputstream.adaptive/assets/65504837/f1e55dda-44a9-4330-8f57-284d7c475375)
![072123164607](https://github.com/xbmc/inputstream.adaptive/assets/65504837/c77dacc2-79a2-4d0d-ad57-3de8e38a3e30)
![072123164619](https://github.com/xbmc/inputstream.adaptive/assets/65504837/5c9ea79c-f81e-4bc5-a133-becd0012f18f)

**Comment by CastagnaIT** [comment]
i think you have to open "SlyGuy common addon" and disable "slyProxy enabled" on advanced settings

**Comment by Ronny-nerd** [comment]
Thanks for the info...CastagnaIT.

**Comment by Ronny-nerd** [comment]
[Manifest-Problem 4 InputStream Adaptive.zip](https://github.com/xbmc/inputstream.adaptive/files/12130768/Manifest-Problem.4.InputStream.Adaptive.zip)

No sound after starting an episode.

![072123183617](https://github.com/xbmc/inputstream.adaptive/assets/65504837/9ae739b8-b039-4cec-9216-5f36e713fb2b)
![072123183634](https://github.com/xbmc/inputstream.adaptive/assets/65504837/f143b020-4415-4fac-ae85-fa9e848de995)
![072123183649](https://github.com/xbmc/inputstream.adaptive/assets/65504837/9dcdc6fb-7714-4a48-9dac-5dd1d446463f)

Sound only came after switching manually.

![072123184257](https://github.com/xbmc/inputstream.adaptive/assets/65504837/7cb21f29-48f9-4b6c-85d8-d92ff6da721e)
![072123184310](https://github.com/xbmc/inputstream.adaptive/assets/65504837/25f672b3-404a-4dab-a022-3ae5edbc2586)

**Comment by Ronny-nerd** [comment]
What surprises and irritates me is that a movie with InputStream Adaptive version 21.1.3 starts and runs without sound problems.

**Comment by CastagnaIT** [comment]
can you stop add screenshots? have no utility and are making this thread a mess
the only thing that could have sense is provide the manifests of a working video in order to try find differences

from what i can see seem there is a wrong url where miss `/r/` on address
thats weird because the code its the same for all

**Comment by CastagnaIT** [comment]
i rechecked whole code around download urls and also tried put manifests on a new test build
but i havent found nothing that can explain this wrong url

@matthuisman are you able to replicate this problem?

**Comment by glennguy** [comment]
@CastagnaIT I see what you're seeing - the variant playlist is in a subfolder of the master playlist, but the segments seem to be calculated against the path of the master instead of the variant (missing /r folder).

**Comment by CastagnaIT** [comment]
exact, also we can see that baseurl on representation is set always from the effective url
https://github.com/xbmc/inputstream.adaptive/blob/Omega/src/parser/HLSTree.cpp#L212
and on download since that segment file have a relative the path, path is joined with the same representation baseurl:
https://github.com/xbmc/inputstream.adaptive/blob/Omega/src/common/AdaptiveStream.cpp#L269-L270
i was thinking of a bug of URL::Join method but testing it given right output

**Comment by MauriceW67** [comment]
> @matthuisman are you able to replicate this problem?

I am not @matthuisman but I can confirm this issue also happens with my setup. I think it has something to do with enabling or disabling Atmos sound in the addon. If I switch audio tracks while playing, I can choose between DD+ and DD+ Atmos, and the first one does produce sound.

Another issue I have (which doesn't happen with Kodi 20.2 and IS:A 20.3.11) is that the subtitles that are selected (Dutch in my case) do not appear.

**Comment by CastagnaIT** [comment]
try test last nightly
https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.adaptive/detail/Omega/93/artifacts
let's see if it's a side effect of memory corruption now fixed

**Comment by MauriceW67** [comment]
@CastagnaIT Unfortunately that test version did not make any difference.

**Comment by Ronny-nerd** [comment]
For me the same result with this build as for MauriceW67.

[Problem InputStream Adaptive 21.1.3 - Kodi Omega - Disney Addon.zip](https://github.com/xbmc/inputstream.adaptive/files/12335777/Problem.InputStream.Adaptive.21.1.3.-.Kodi.Omega.-.Disney.Addon.zip)

**Comment by GG-76** [comment]
Hi, I have the same version of both Kodi (Omega) and InputStream (21.1.3). When playing some series on Disney+, I have the same problem, a "broken" picture, non-functional Dolby Atmos sound, subtitles not appearing in my language, etc. At the same time, but when I play, for example, the movie Guardians of the Galaxy 3, which has the same parameters, there is no problem at all. Also, the add-on Osmosis for exporting streams to the Kodi library stopped working correctly for me with Disney+ (not completely, but for example when I wanted to add the new High School Musical series), which may not be related, but there was no problem with this add-on before, and it still works with other add-ons without problems.

**Comment by CastagnaIT** [comment]
@Ronny-nerd @MauriceW67 can you give me a feedback with following test build?

PS. is required last Kodi nightly due to API change

**Comment by CastagnaIT** [comment]
i made a mistake, this is the new test build url: https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.adaptive/detail/PR-1364/2/artifacts

**Comment by MauriceW67** [comment]
@CastagnaIT I justed tested this build with the latest kodi nightly and the Disney+ addon.

My findings sofar:

- Video plays fine in 4K / Dolby Vision.
- Audio plays, but I have to manually select the correct stream. If I enable Atmos in the addon, it does choose the Atmos track (showing as 16 channels), but my receiver stays silent. If I disable Atmos in the addon, it defaults to an English AAC 2-channel track which also does not give any sound. Manually selecting the DD+ track does produce sound correctly.
- Subtitles are a strange thing. Whatevery I do, subtitles in my own language (Dutch) will not show, even though they show as being enabled in Kodi. Switching to another language works, but switching back to Dutch makes them disappear again.

Btw, it doesn't mattter if I setup default and allowed audio and subtitle languages within the addon or not. The odd behavior with the sound and subtitle tracks stays the same. Maybe @matthuisman can shed some light on this?

Please note that with Kodi 20.2 and IS:A 20.3.11 the issues with the audio and subtitles streams are not there.

If you need any logs, let me know.

**Comment by CastagnaIT** [comment]
the most important thing is you need to make sure that manifest alteration proxy setting is set disabled from slyshguy addon

the default audio/subtitle track choosed, not depends from ISAdaptive
but without log what could i do, nothing... since any maintainer here dont have D+
it's already a mess to be able to figure it out without being able to replicate locally

the only thing i can theorize is that its possible that the log there are same errors due to same wrong url path without `/r` for unknown reason
in this case maybe having a log with enabled Curl logging from kodi logging component settings maybe could help show something more

**Comment by matthuisman** [comment]
@MauriceW67 
Just focus on the single sound issue this ticket is for.

Confirm that is fixed with the test builds and still broken with current stable builds.

You can turn off slyguy proxy in the slyguy common settings (under services)

**Comment by Ronny-nerd** [comment]
@matthuisman ...What about your test on this subject?

Have the same problems with your new test build from today...CastagnaIT.

[Problem.zip](https://github.com/xbmc/inputstream.adaptive/files/12443951/Problem.zip)

**Comment by CastagnaIT** [comment]
the log seem bit better but
```
2023-08-26 00:10:38.569 T:19685   error <general>: AddOnLog: inputstream.adaptive: Open: CURLOpen failed
2023-08-26 00:10:38.569 T:19685   error <general>: AddOnLog: inputstream.adaptive: [AS-259] Download failed, internal error: b07ff016-7c04-4915-9990-581f5620c3c5/524c-MAIN/03/audio_ec3_256K/map.mp4a
```
however again a bad path
i found something e.g.

```
#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID="atmos",NAME="Deutsch",LANGUAGE="de",AUTOSELECT=YES,CHANNELS="6",URI="r/composite_256k_ec-3_de_PRIMARY_a39c570a-3c46-4db6-b42f-751c28f17566_10310d0e-3231-46ca-8ec4-822aec4d1f6b.m3u8"

#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID="eac-3",NAME="Deutsch",LANGUAGE="de",AUTOSELECT=YES,CHANNELS="6",URI="r/composite_256k_ec-3_de_PRIMARY_a39c570a-3c46-4db6-b42f-751c28f17566_10310d0e-3231-46ca-8ec4-822aec4d1f6b.m3u8"
```

two renditions that point to the same manifest, now i added code to ignore duplicates that i havent taken in account on rework
the manifest call atmos a track that isnt atmos...good mixed things by luck at least the "joc" param is missing so not a problem

i noticed also that there same audio tracks with multiple bitrate like:
```
#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID="aac-128k",NAME="Español",LANGUAGE="es-ES",AUTOSELECT=YES,CHANNELS="2",URI="r/composite_128k_mp4a.40.2_es-ES_PRIMARY_017d7b89-cde0-49c9-964e-83887fdc30e0_4d3762ef-ee9a-4835-b76f-1eb9ec83ecb1.m3u8"

#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID="aac-64k",NAME="Español",LANGUAGE="es-ES",AUTOSELECT=YES,CHANNELS="2",URI="r/composite_64k_mp4a.40.2_es-ES_PRIMARY_d3fd986b-cdf6-4e01-bae2-1202a4e5539f_4d3762ef-ee9a-4835-b76f-1eb9ec83ecb1.m3u8"
```
this will be a bit problem for quality auto selection since not exist a BANDWIDTH or BITRATE HLS param
i think will be choosen the first in order of apparence in the manifest
however this not concern the current issue

Jenkins CI builds are currently dead on the repository
i will prepare a test build manually soon

**Comment by CastagnaIT** [comment]
here new test
[inputstream.adaptive-android-aarch64-omega-decc930_1.zip](https://github.com/xbmc/inputstream.adaptive/files/12444846/inputstream.adaptive-android-aarch64-omega-decc930_1.zip)
please disable Curl on logging its not more needed, also manifests are not more needed, just log file

**Comment by Ronny-nerd** [comment]
Here is the Kodi Debug LOG file with your latest test build inputstream.adaptive-android-aarch64-omega-decc930_1.zip

[Problem2.zip](https://github.com/xbmc/inputstream.adaptive/files/12445116/Problem2.zip)

The problem persists, even with InputStream Adaptive version 21.2.0

**Comment by MauriceW67** [comment]
I also just tested with the decc930_1 version.

SlyProxy is disabled and Atmos is enabled in both Slyguy Common and the Disney+ addon.

When starting an episode, the addon selects the "DD+ ATMOS 16-channel" audio track, but no sound is played (note: the 3-second Disney+ bumper before the episode does play sound).

Manually switching to the normal DD+ track with 6 channels makes the sound play.

Debug log here:

[kodi.log](https://github.com/xbmc/inputstream.adaptive/files/12445205/kodi.log)

With Atmos disabled, the behavior is slightly different. The addon then selects the AAC-LC 2-channel audio track as default, but no sound can be heard. Switching manually to the DD+ 6-channel track makes the sound play again.

Debug log here:

[kodi.log](https://github.com/xbmc/inputstream.adaptive/files/12445224/kodi.log)

EDIT: Just noticed a second annoying issue. If the episode is played for a second time with Atmos disabled, the previous manually selected DD+ 6-channel track is apparently remembered, but no sound can be heard. Switching to the AAC-LC track that previously didn't play sound, now does play sound. But switching back to DD+ 6-channel makes everything go quiet again. So it seems that the audio track that is chosen first by the addon, never produces any sound, no matter if you switch back and forth to it.

**Comment by CastagnaIT** [comment]
ok i found how to rougly reproduce it
i think finally found the problem was on periods
the current code creates periods from EXT-X-DISCONTINUITY when parsing for the first time any child manifest
but the code set the baseurl to current representation only, then when switch to another period miss to set baseurl
the current code work flow is a bit messy also to understand...

test build, should works:
[inputstream.adaptive-android-aarch64-omega-a16815f_1.zip](https://github.com/xbmc/inputstream.adaptive/files/12445434/inputstream.adaptive-android-aarch64-omega-a16815f_1.zip)

**Comment by Ronny-nerd** [comment]
inputstream.adaptive-android-aarch64-omega-a16815f_1.zip  = Top...thanks CastagnaIT ...the new build now works.

**Comment by MauriceW67** [comment]
New build works for me too. And subtitles also, so the other issue I created can also be resolved.

Thanks for all the work you are putting into this @CastagnaIT

**Comment by GG-76** [comment]
@CastagnaIT Hi, I would like to ask if this is fixed in version 21.2.0. because I can't install the fix file here (it's for aarch64). Unfortunately, on the latest Nightly Kodi Omega with version 21.2.0, the video and subtitles work for me when playing from Disney+, but I still don't get the Atmos sound :( It only plays the first few seconds at the Disney+ logo and then nothing... Thank you

**Comment by matthuisman** [comment]
the issue was fixed with PR: https://github.com/xbmc/inputstream.adaptive/pull/1364
It was merged 5 days ago.

There is no new release yet that includes it.

But, you can just download the build of latest master from below:
https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.adaptive/detail/Omega/100/artifacts
